src/main/java/uk/gov/hmcts/cmc/claimstore/services/DocumentManagementService.java

```diff
@@ -0,0 +1,64 @@
+package uk.gov.hmcts.cmc.claimstore.services;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.core.io.Resource;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import uk.gov.hmcts.cmc.claimstore.exceptions.DocumentManagementException;
+import uk.gov.hmcts.document.DocumentDownloadClientApi;
+import uk.gov.hmcts.document.DocumentMetadataDownloadClientApi;
+import uk.gov.hmcts.document.DocumentUploadClientApi;
+import uk.gov.hmcts.document.domain.Document;
+import uk.gov.hmcts.document.domain.UploadResponse;
+import uk.gov.hmcts.document.utils.InMemoryMultipartFile;
+
+import java.net.URI;
+
+import static java.util.Collections.singletonList;
+
+@Service
+public class DocumentManagementService {
+
+    static final String APPLICATION_PDF = "application/pdf";
+    static final String PDF_EXTENSION = ".pdf";
+    static final String FILES_NAME = "files";
+    private final DocumentMetadataDownloadClientApi documentMetadataDownloadClient;
+    private final DocumentDownloadClientApi documentDownloadClient;
+    private final DocumentUploadClientApi documentUploadClient;
+
+    @Autowired
+    public DocumentManagementService(
+        final DocumentMetadataDownloadClientApi documentMetadataDownloadApi,
+        final DocumentDownloadClientApi documentDownloadClientApi,
+        final DocumentUploadClientApi documentUploadClientApi
+    ) {
+        this.documentMetadataDownloadClient = documentMetadataDownloadApi;
+        this.documentDownloadClient = documentDownloadClientApi;
+        this.documentUploadClient = documentUploadClientApi;
+    }
+
+    public String uploadDocument(final String authorisation, final String originalFileName,
+                                 final byte[] documentBytes, final String contentType) {
+        final MultipartFile file = new InMemoryMultipartFile(FILES_NAME, originalFileName, contentType, documentBytes);
+        final UploadResponse response = documentUploadClient.upload(authorisation, singletonList(file));
+
+        final Document document = response.getEmbedded().getDocuments().stream()
+            .findFirst()
+            .orElseThrow(() ->
+                new DocumentManagementException("Document management failed uploading file" + originalFileName));
+
+        return URI.create(document.links.self.href).getPath();
+    }
+
+    public byte[] downloadDocument(final String authorisation, final String documentSelfPath) {
+        final Document documentMetadata = documentMetadataDownloadClient.getDocumentMetadata(authorisation,
+            documentSelfPath);
+
+        final ResponseEntity<Resource> responseEntity = documentDownloadClient.downloadBinary(authorisation,
```

Comment:
I would have preferred you to assign `Resource` as variable in one line and then call `.getByteArray()` on it in another line. Currently we have read of the HTTP response split into two lines whether we could have it on one line ;)

Reply:
done